app/javascript/utils/index.ts

```diff
@@ -77,3 +77,72 @@ export const clickTracking = (affiliate: string, module: string, query: string,
     })
   });
 };
+
+const getLuminance = (red: number, green: number, blue: number) => {
+  const rgb = [red, green, blue].map((index) => {
+    const value = index / 255;
+    return value <= 0.03928
+      ? value / 12.92
+      : Math.pow((value + 0.055) / 1.055, 2.4);
+  });
+  return rgb[0] * 0.2126 + rgb[1] * 0.7152 + rgb[2] * 0.0722;
+};
+
+const getRgbColorObject = (color: string) => {
+  const rgbStrLen = 'rgb('.length;
+  const colorArr = color.substring(rgbStrLen, color.lastIndexOf(')')).split(', ');
+  return {
+    red: parseInt(colorArr[0], 10),
+    green: parseInt(colorArr[1], 10),
+    blue: parseInt(colorArr[2], 10)
+  };
+};
+
+/*
+  AA-level small text: ${contrastRatio < 1/4.5 ? 'PASS' : 'FAIL' }
+  AAA-level small text: ${contrastRatio < 1/7 ? 'PASS' : 'FAIL' }
+  AAA-level large text: ${contrastRatio < 1/4.5 ? 'PASS' : 'FAIL' }
+  AA-level large text: ${contrastRatio < 1/3 ? 'PASS' : 'FAIL' }
+*/
+const calculateContrastRatio = (bgColor: string, fgColor: string) => {
+  const color1rgb = getRgbColorObject(fgColor);
+  const color2rgb = getRgbColorObject(bgColor);
+
+  // calculate the relative luminance
+  const color1luminance = getLuminance(color1rgb.red, color1rgb.green, color1rgb.blue);
+  const color2luminance = getLuminance(color2rgb.red, color2rgb.green, color2rgb.blue);
+
+  // calculate the color contrast ratio
+  const ratio = color1luminance > color2luminance 
+    ? ((color2luminance + 0.05) / (color1luminance + 0.05))
+    : ((color1luminance + 0.05) / (color2luminance + 0.05));
+
+  return ratio;
+};
+
+interface colorContrastItemProps {
+  backgroundItemClass: string, 
+  foregroundItemClass: string,
+  isForegroundItemBtn?: boolean
+}
+
+export const checkColorContrast = ({ backgroundItemClass, foregroundItemClass, isForegroundItemBtn = false }: colorContrastItemProps) => {
```

Comment:
We should rename this function since it does more than just check the color contrast.